lib/src/emitter.dart

```diff
@@ -336,6 +337,62 @@ class DartEmitter extends Object
     return out;
   }
 
+  @override
+  StringSink visitExtensionType(ExtensionType spec, [StringSink? output]) {
+    final out = output ??= StringBuffer();
+    spec.docs.forEach(out.writeln);
+    for (var a in spec.annotations) {
+      visitAnnotation(a, out);
+    }
+
+    out.write('extension type ');
+    if (spec.constant) out.write('const ');
+    out.write(spec.name);
+    visitTypeParameters(spec.types.map((r) => r.type), out);
+    if (spec.primaryConstructorName.isNotEmpty) {
+      out.write('.${spec.primaryConstructorName}');
+    }
+    out.write('(');
+    _visitRepresentationDeclaration(spec.representationDeclaration, out);
+    out.write(')');
+
+    if (spec.implements.isNotEmpty) {
+      out
+        ..write(' implements ')
+        ..writeAll(
+            spec.implements.map<StringSink>((m) => m.type.accept(this)), ', ');
```

Comment:
consider joining with ', ' instead so we don't have a trailing comma?

Reply:
It is indeed what it does - it takes in a separator between any two elements. I do add a space here to make the non-formattable code look a bit better, but maybe I shouldn't for consistency and just wait for the new version of dart_style.